internal/pkg/deploy/cloudformation/stack/lb_web_app.go

```diff
@@ -0,0 +1,199 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package stack
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/addons"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/archer"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/manifest"
+	"github.com/aws/amazon-ecs-cli-v2/internal/pkg/template"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/cloudformation"
+)
+
+// Template rendering configuration.
+const (
+	lbWebAppTemplateName              = "lb-web-app"
+	lbWebAppRulePriorityGeneratorPath = "custom-resources/alb-rule-priority-generator.js"
+)
+
+// Parameter logical IDs unique for a load balanced web application.
+const (
+	LBWebAppHTTPSParamKey           = "HTTPSEnabled"
+	LBWebAppContainerPortParamKey   = "ContainerPort"
+	LBWebAppRulePathParamKey        = "RulePath"
+	LBWebAppHealthCheckPathParamKey = "HealthCheckPath"
+)
+
+type templater interface {
+	Template() (string, error)
+}
+
+// LoadBalancedWebApp represents the configuration needed to create a CloudFormation stack from a load balanced web application manifest.
+type LoadBalancedWebApp struct {
+	*app
+	manifest     *manifest.LoadBalancedWebApp
+	httpsEnabled bool
+
+	addons templater
+}
+
+// NewLoadBalancedWebApp creates a new LoadBalancedWebApp stack from a manifest file.
+// The stack is configured
+func NewLoadBalancedWebApp(mft *manifest.LoadBalancedWebApp, env, proj string, rc RuntimeConfig) (*LoadBalancedWebApp, error) {
+	parser := template.New()
+	addons, err := addons.New(mft.Name)
+	if err != nil {
+		return nil, fmt.Errorf("new addons: %w", err)
+	}
+	return &LoadBalancedWebApp{
+		app: &app{
+			name:    mft.Name,
+			env:     env,
+			project: proj,
+			tc:      mft.ApplyEnv(env).TaskConfig,
+			rc:      rc,
+			parser:  parser,
+		},
+		manifest:     mft,
+		httpsEnabled: false,
+
+		addons: addons,
+	}, nil
+}
+
+// NewHTTPSLoadBalancedWebApp  creates a new LoadBalancedWebApp stack from its manifest that needs to be deployed to
+// a environment within a project. It creates an HTTPS listener and assumes that the environment
+// it's being deployed into has an HTTPS configured listener.
+func NewHTTPSLoadBalancedWebApp(mft *manifest.LoadBalancedWebApp, env, proj string, rc RuntimeConfig) (*LoadBalancedWebApp, error) {
+	parser := template.New()
```

Comment:
I guess you could also call `NewLoadBalancedWebApp` and set httpsEnabled to be true on the return struct :shrug: